website/src/controllers/groups.js

```diff
@@ -518,10 +518,12 @@ api.leave = function(req, res, next) {
   // When removing the user from challenges, should we keep the tasks?
   var keep = (/^remove-all/i).test(req.query.keep) ? 'remove-all' : 'keep-all';
 
-  group.leave(user, keep, function(err){
-    if (err) return next(err);
-    user = group = keep = null;
-    return res.send(204);
+  Group.findOne({"_id": group._id}).populate('members').exec(function (err, group) {
```

Comment:
@TheHollidayInn are we sure this has to be fetched once again? We already have a group object from the db, and ids don't need to be populated